consensus/src/state_computer.rs

```diff
@@ -155,18 +167,21 @@ impl StateComputer for ExecutionProxy {
         let skip_clean = blocks.is_empty();
         let mut latest_epoch: u64 = 0;
         let mut latest_round: u64 = 0;
+        let mut payloads = Vec::new();
 
         for block in blocks {
             block_ids.push(block.id());
-            txns.extend(block.transactions_to_commit(&self.validators.lock()));
+            if block.block().payload().is_some() {
+                payloads.push(block.block().payload().unwrap().clone());
+            }
+            debug!("QSE: getting data in commit, round {}", block.round());
+            let signed_txns = self.data_manager.get_data(block.block()).await?;
```

Comment:
Do we have timers (monitor!) for get_data? Seems like we want one

Reply:
Sure, why not.